### PR TITLE
chore(ci/docker): enable cross-platform builds (closes #132)

### LINKED ISSUE #132: Semantic versioning and cross-architecture support for Docker images
## Feature Request

Currently, the only available image on Docker Hub is tagged 'latest', perhaps the project could follow semver & the image tags could follow suit.

### PR DESCRIPTION
**Pull request by ChipWolf**

### CODE REVIEW

File: .github/workflows/build-and-deploy.yml [hunk 1]
```diff
@@ -1,19 +1,60 @@
 name: Build and deploy on push
 
 on:
+  workflow_dispatch:
   push:
     branches:
+      - dev
```
Comment (nimarion) [inline .github/workflows/build-and-deploy.yml:7]: Change dev to development to match the current repo.
Also change main to master. Ill change it when i  rename everything to main.

**Review by nimarion — CHANGES_REQUESTED** [review]
Looks good. Just have a look at the branch names. 
I appreciate your contribution to this project 🎉

**Comment by ChipWolf** [comment]
All sorted @nimarion  🚀

**Comment by ChipWolf** [comment]
Only issue with this, the [base image](https://hub.docker.com/layers/maven/library/maven/3.6.3-jdk-14/images/sha256-68c92e3e45b0cfd6d3903f77bd972a6ecc9665191c7bf88aabd7ba1878d9f96d?context=explore) you're [currently using](https://github.com/nimarion/promcord/blob/master/Dockerfile#L1) for maven only has a `linux/amd64` build

**Comment by ChipWolf** [comment]
Hey @nimarion, could you review this once again, please?

File: .github/workflows/build-and-deploy.yml [hunk 2]
```diff
@@ -1,19 +1,60 @@
...
+
+      - name: Build and Push Docker Image
+        if: github.ref == 'refs/heads/development' || github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/')
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          context: .
+          file: ./Dockerfile
+          platforms: linux/386,linux/amd64,linux/arm/v6,linux/arm/v7,linux/arm64/v8
```
Comment (nimarion) [inline .github/workflows/build-and-deploy.yml:59]: Do i need any changes here if we change the base image to 3.8.1-adoptopenjdk-16?

Comment (ChipWolf) [reply .github/workflows/build-and-deploy.yml:59]: As long as all of the platforms from the list in the workflow are provided by your base image, you should be good to go.
I would review the platforms lists on changing the base image.

Comment (nimarion) [reply .github/workflows/build-and-deploy.yml:59]: Alright. I'm going to change it in #96 to 3.8.1-adoptopenjdk-16 👍

Comment (nimarion) [reply .github/workflows/build-and-deploy.yml:59]: I had a look at the Dockerfile and wonderd if the openjdk image in line 6 is going to cause any issues?

Comment (ChipWolf) [reply .github/workflows/build-and-deploy.yml:59]: I can't say I fully understand why line 6 exists, the JDK is in the maven image. I'll run a few test builds

Comment (nimarion) [reply .github/workflows/build-and-deploy.yml:59]: Thats a multitstage docker build to reduce the image size

**Pull request closed without merging**